### PR TITLE
Add back Ollama integration

### PR DESCRIPTION
**Pull request by Kheiden**
Tests are passing.

## Release Notes

- **New Features**
  - Introduced a new environment variable `OLLAMA_HOST_URL` for configuration.
  - Added a new service `simplemind-test` in the Docker setup to enhance testing capabilities.
  - Expanded provider options by including `Ollama` for API interactions.

- **Documentation**
  - Updated `README.md` with clearer instructions on building the project and using Docker containers.

- **Bug Fixes**
  - Enhanced unit tests for the `ollama` provider to ensure robust functionality.

- **Chores**
  - Updated `.gitignore` to exclude `requirements.txt` from version control.

### CODE REVIEW

**Review by kennethreitz — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -9,4 +9,4 @@ WORKDIR /src
 
 RUN pip install -r requirements.txt
 
-ENTRYPOINT ["python", "build.py"]
\ No newline at end of file
+ENTRYPOINT ["python", "main.py"]
```
[the remark this answers is not in the document]
Comment (Kheiden) [reply Dockerfile:12]: COPY is unnecessary as the docker-compose file binds a host volume

**Comment by Siddhesh-Agarwal** [comment]
Hello @Kheiden, we just added a `BaseProvider` class (in `simplemind/models.py`), can you ensure that the Ollama provider inherits it?

**Comment by Kheiden** [comment]
Hi @Siddhesh-Agarwal, It appears that there's a circular dependency the `BaseProvider` class:
```
simplemind-test-1  | Traceback (most recent call last):
simplemind-test-1  |   File "/src/main.py", line 2, in <module>
simplemind-test-1  |     import simplemind as sm                                                                                                                                                 
simplemind-test-1  |   File "/src/simplemind/__init__.py", line 1, in <module>                                                                                                                   
simplemind-test-1  |     from .models import Conversation                                                                                                                                        
simplemind-test-1  |   File "/src/simplemind/models.py", line 7, in <module>                                                                                                                     
simplemind-test-1  |     from .utils import find_provider                                                                                                                                        
simplemind-test-1  |   File "/src/simplemind/utils.py", line 1, in <module>                                                                                                                      
simplemind-test-1  |     from .providers import providers                                                                                                                                        
simplemind-test-1  |   File "/src/simplemind/providers/__init__.py", line 1, in <module>                                                                                                         
simplemind-test-1  |     from .anthropic import Anthropic
simplemind-test-1  |   File "/src/simplemind/providers/anthropic.py", line 6, in <module>                                                                                                        
simplemind-test-1  |     from simplemind.models import BaseProvider, Conversation, Message                                                                                                       
simplemind-test-1  | ImportError: cannot import name 'BaseProvider' from partially initialized module 'simplemind.models' (most likely due to a circular import) (/src/simplemind/models.py)     
Gracefully stopping... (press Ctrl+C again to force)   
```

**Comment by Siddhesh-Agarwal** [comment]
@Kheiden thanks for bringing it to my attention. I'll try and fix it.

**Comment by Kheiden** [comment]
I created PR#10 to fix the issue for ollama. The other providers may need a call to the parent class constructor.

**Comment by kennethreitz** [comment]
Replaced by the other PR, yes?

**Pull request closed without merging**